### PR TITLE
[Numpy] numpy compatible invert

### PR DESCRIPTION
**Pull request by mikemwx**
## Description ##
- add numpy compatible [invert](https://docs.scipy.org/doc/numpy/reference/generated/numpy.invert.html) 
- add a helper function for checking if input or output is integer

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- add invert and its test
- a helper function to check for checking if input and output is of type integer at ` src/operator/operator_common.h type_is_int`

## Comments ##
- pass CPU,GPU test, pylint, cpplint
- Thank @haojin2 for reviewing

### CODE REVIEW

**Comment by roywei** [comment]
@mxnet-label-bot add [numpy]

File: python/mxnet/ndarray/numpy/_op.py [hunk 1]
```diff
@@ -1414,6 +1413,48 @@ def rint(x, out=None, **kwargs):
     """
     return _unary_func_helper(x, _npi.rint, _np.rint, out=out, **kwargs)
 
+def invert(x, out=None, **kwargs):
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: One more blank line above.

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -1414,6 +1413,48 @@ def rint(x, out=None, **kwargs):
...
+    This function differs from the original `numpy.invert
+    <https://docs.scipy.org/doc/numpy/reference/generated/numpy.invert.html>`_ in
+    the following way(s):
+
+    - only ndarray or scalar is accpted as valid input, tuple of ndarray is not supported
+    - broadcasting to `out` of different shape is currently not supported
+    - when input is plain python numerics, the result will not be stored in the `out` param
+
+    """
+    return _unary_func_helper(x, _npi.invert, _np.invert, out=out, **kwargs)
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: One more blank line below

Comment (haojin2) [reply python/mxnet/ndarray/numpy/_op.py]: Same for all python functions, they all shall be 2 blank lines apart.

Comment (mikemwx) [reply python/mxnet/ndarray/numpy/_op.py]: Thank you. Resolved.

**Pull request closed without merging**